packages/flutter/lib/src/foundation/timeline.dart

```diff
@@ -0,0 +1,395 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:developer';
+import 'dart:typed_data';
+
+import 'package:meta/meta.dart';
+
+import '_timeline_io.dart'
+  if (dart.library.js_util) '_timeline_web.dart' as impl;
+
+/// Measures how long blocks of code take to run.
+///
+/// This class can be used as a drop-in replacement for [Timeline] as it
+/// provides methods compatible with [Timeline] signature-wise, and it has
+/// minimal overhead.
+///
+/// Provides [reset] and [collect] methods that make it convenient to use in
+/// frame-oriented environment where collected metrics can be attributed to a
+/// frame, then aggregated into frame statistics, e.g. frame averages.
+///
+/// Forwards measurements to [Timeline] so they appear in Flutter DevTools.
+abstract final class FlutterTimeline {
+  static _BlockBuffer _buffer = _BlockBuffer();
```

Comment:
I am concerned that all this code for non-release functionality is actually unnecessarily retained in release mode and blowing up binary size there.

Reply:
You may have to guard these more aggressively behind `!kReleaseMode` to make sure it is properly removed. I believe we also have a test that makes sure some things don't make it into release mode, can you please add some of the concepts introduced by this PR there?

Reply:
Where do I find that test?